### PR TITLE
Update Compose and others

### PR DESCRIPTION
**Pull request by djnotes**
Compose to 1.3.0
Filament to  1.28.2
Android Gradle plugin to 8.0.0-alpha07
Kotlin Gradle plugin to 1.7.10
Gradle to 7.5
In Viewer.kt, change the signature of onViewAttachedToWindow and onViewDetachedFromWindow to use android.view.View rather than android.view.View? because the former didn't compile with error "x does not implement y"
Other minor library updates

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+* text=auto
```
Comment (romainguy) [inline .gitattributes:1]: Please don't change the .git files

File: .gitignore [hunk 2]
```diff
@@ -15,3 +15,4 @@
 .externalNativeBuild
 .cxx
 local.properties
+filament/bin/matc.exe
```
Comment (romainguy) [inline .gitignore:18]: Keep next to the other filament/bin

File: app/build.gradle [hunk 3]
```diff
@@ -55,27 +56,27 @@ android {
...
-        freeCompilerArgs += "-Xopt-in=kotlin.RequiresOptIn"
+        freeCompilerArgs += "-opt-in=kotlin.RequiresOptIn"
     }
 }
 
 dependencies {
-    implementation "androidx.core:core-ktx:1.6.0"
-    implementation 'androidx.appcompat:appcompat:1.4.0-beta01'
+    implementation "androidx.core:core-ktx:1.9.0"
+    implementation 'androidx.appcompat:appcompat:1.7.0-alpha01'
```
Comment (romainguy) [inline app/build.gradle:65]: Why use an alpha here?

File: build.gradle [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+        compose_version = '1.3.0'
     }
     repositories {
         google()
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.3'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.31"
+        classpath 'com.android.tools.build:gradle:8.0.0-alpha07'
```
Comment (romainguy) [inline build.gradle:12]: Let's not depend on previews of Android Studio

**Review by romainguy — CHANGES_REQUESTED** [review]

**Comment by djnotes** [comment]
Rolled things back to stable and other stuff

File: .idea/codeStyles/Project.xml [hunk 5]
```diff
@@ -0,0 +1,123 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (romainguy) [inline .idea/codeStyles/Project.xml:1]: Please nuke this file. It's changing the code style settings for this project

**Review by romainguy — APPROVED** [review]
Thank you!

**Pull request merged**